src/codemagic/utilities/backwards_file_reader.py

```diff
@@ -1,39 +1,51 @@
 import io
 from pathlib import Path
 from typing import Iterable
+from typing import Optional
+from typing import TextIO
 from typing import Union
 
 
-def iter_backwards(file_path: Union[Path, str], buffer_size=8192) -> Iterable[str]:
+def _iter_backwards(file_descriptor: TextIO, buffer_size: int, file_size: Optional[int] = None):
+    current_segment = None
+    offset_from_end = 0
+    unprocessed_size = file_descriptor.seek(0, io.SEEK_END)
+    if file_size is None:
+        file_size = unprocessed_size
+
+    while unprocessed_size > 0:
+        offset_from_end = min(file_size, offset_from_end + buffer_size)
+        file_descriptor.seek(file_size - offset_from_end)
```

Comment:
Sadly this cannot be used as not all `io` types support seeking like that and this would result in
```python
io.UnsupportedOperation: can't do nonzero end-relative seeks
```